Archive/Imo/Imo1982Q1.lean

```diff
@@ -0,0 +1,131 @@
+/-
+Copyright (c) 2024 Alex Brodbelt. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex Brodbelt
+-/
+
+import Mathlib.Tactic
+/-!
+# IMO 1982 Q1
+
+The function $f(n)$ is defined for all postive integers $n$ and takes on
+non-negative integer values. Also, for all m, n
+
+$f (m + n) - f(m) - f(n) = 0 \text{ or } 1$
+$f(2) = 0, f(3) > 0, and f(9999) = 3333.$
+
+Determine f(1982).
+
+The solution is based on the one at the
+[Art of Problem Solving](https://artofproblemsolving.com/wiki/index.php/1982_IMO_Problems/Problem_1)
+website.
+-/
+
+namespace Imo1982Q1
+
+structure IsGood (f : ℕ → ℕ) : Prop where
```

Comment:
You should state the problem in terms of `f : ℕ+ → ℕ`. We are trying to be formal here, aren't we?

Reply:
I don't see why proving inequalities on `ℕ+` would be any more complicated. If you have any questions, feel free to ask on Zulip.